### PR TITLE
aaa: init at 1.1.1

### PR DESCRIPTION
**Pull request by asciimoth**
###### Description of changes

Add [aaa](https://github.com/DomesticMoth/aaa) - terminal viewer for 3a format.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by DomesticMoth** [comment]
TC @balsoft

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -15939,4 +15939,13 @@
     github = "wuyoli";
     githubId = 104238274;
   };
+  DomesticMoth = {
```
Comment (figsoda) [inline maintainers/maintainer-list.nix:15942]: please keep the file in alphabetical order and put this in a separate commit: `maintainers: add DomesticMoth`

File: pkgs/tools/misc/aaa/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{ lib
+, stdenv
```
Comment (figsoda) [inline pkgs/tools/misc/aaa/default.nix:2]:
```suggestion
```

File: pkgs/tools/misc/aaa/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-gIOlPjZOcmVLi9oOn4gBv6F+3Eq6t5b/3fKzoFqxclw=";
+  };
+  cargoSha256 = "sha256-ugB0r9qiGRurc30GrJH4MKM6fWZ99+f1Gy7/1lSmrwU=";
+
+  meta = with lib; {
+    description = "Terminal viewer for 3a format";
+    homepage = "https://github.com/DomesticMoth/aaa";
+    license = with licenses; [ gpl3 ];
```
Comment (figsoda) [inline pkgs/tools/misc/aaa/default.nix:22]:
```suggestion
    license = with licenses; [ gpl3Only ];
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -3089,6 +3089,8 @@ with pkgs;
 
   bat-extras = recurseIntoAttrs (callPackages ../tools/misc/bat-extras { });
 
+  aaa = callPackage ../tools/misc/aaa {};
```
Comment (figsoda) [inline pkgs/top-level/all-packages.nix:3092]: please keep this in (mostly) alphabetical order

**Review by figsoda — CHANGES_REQUESTED** [review]
just a couple nitpicks

**Review by figsoda — APPROVED** [review]
